server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -239,15 +246,45 @@ public InternalEngine(EngineConfig engineConfig) {
     }
 
     private LocalCheckpointTracker createLocalCheckpointTracker(
-        BiFunction<Long, Long, LocalCheckpointTracker> localCheckpointTrackerSupplier) throws IOException {
-        final long maxSeqNo;
-        final long localCheckpoint;
-        final SequenceNumbers.CommitInfo seqNoStats =
-            SequenceNumbers.loadSeqNoInfoFromLuceneCommit(store.readLastCommittedSegmentsInfo().userData.entrySet());
-        maxSeqNo = seqNoStats.maxSeqNo;
-        localCheckpoint = seqNoStats.localCheckpoint;
-        logger.trace("recovered maximum sequence number [{}] and local checkpoint [{}]", maxSeqNo, localCheckpoint);
-        return localCheckpointTrackerSupplier.apply(maxSeqNo, localCheckpoint);
+        BiFunction<Long, Long, LocalCheckpointTracker> localCheckpointTrackerSupplier) {
```

Comment:
any chance we can make this method static and give it what it needs? I'm afraid we'll use an uninitialized field by mistake (as we call it from the constructor).

Reply:
+1. I pushed https://github.com/elastic/elasticsearch/pull/34474/commits/779c5c7db1c48a4d9e4a13b591150de09fbd7cc3.